### PR TITLE
chore: migrated linting from tslint to eslint

### PR DESCRIPTION
**Pull request by patelvimal**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

```
[ ] Yes
[ ] No
```

### CODE REVIEW

**Comment by LayZeeDK** [comment]
Things to consider when reviewing and accepting this PR:
1. Our GitHub workflows are currently broken. @NachoVazquez is working on a fix.
2. If we can run angular-eslint locally and get rid of all lint errors, we will be willing to merge your PR even before (1) is fixed.
3. Mostly, default rules would be good. But we are using a few opinionated lint settings such as member ordering and `readonly` requirement.

**Comment by NachoVazquez** [comment]
Hi, @patelvimal sorry for the delay in reviewing this PR. I'll take a look tomorrow. Thanks for contributing.

File: libs/ngworker/lumberjack/console-driver/src/lib/configuration/lumberjack-console-driver-root.module.ts [hunk 1]
```diff
@@ -39,12 +39,11 @@ export function consoleDriverFactory(
 })
 export class LumberjackConsoleDriverRootModule {
   constructor(
-    // tslint:disable: no-any no-null-keyword
+    // eslint-disable-next-line  @typescript-eslint/no-non-null-assertion
```
Comment (LayZeeDK) [inline libs/ngworker/lumberjack/console-driver/src/lib/configuration/lumberjack-console-driver-root.module.ts:42]: issue: This conversion doesn't match the previous setting. It should have an `eslint-disable` as well as an `eslint-enable` comment.

File: libs/ngworker/lumberjack/http-driver/src/lib/configuration/lumberjack-http-driver-root.module.ts [hunk 2]
```diff
@@ -42,12 +42,10 @@ export function httpDriverFactory(
 })
 export class LumberjackHttpDriverRootModule {
   constructor(
-    // tslint:disable: no-any no-null-keyword
     @Optional()
     @SkipSelf()
     @Inject(LumberjackHttpDriverRootModule)
     maybeNgModuleFromParentInjector: LumberjackHttpDriverRootModule = null as any
```
Comment (LayZeeDK) [inline libs/ngworker/lumberjack/http-driver/src/lib/configuration/lumberjack-http-driver-root.module.ts:48]: issue: We need to disable the no null and no any lint rules for these lines

File: libs/ngworker/lumberjack/src/lib/formatting/lumberjack-format-log.spec.ts [hunk 3]
```diff
@@ -10,6 +10,7 @@ import { LumberjackTimeService } from '../time/lumberjack-time.service';
 import { lumberjackFormatLog } from './lumberjack-format-log';
 
 function parseFormattedLog(formattedLog: string) {
+  // eslint-disable-next-line
   const formattedLogPattern = /^([a-z]+) ([0-9\.:\-TZ]+) (\[(.+)\] )?(.*)$/;
   const [_, level, timestamp, taggedScopeWithEndingSpace = '', scope = '', message] =
```
Comment (LayZeeDK) [inline libs/ngworker/lumberjack/src/lib/formatting/lumberjack-format-log.spec.ts:15]: issue: Lint error for the `_` variable.

**Review by LayZeeDK — CHANGES_REQUESTED** [review]
Remaining lint errors, see files with check annotations under the *Files changed* tab.

File: libs/ngworker/lumberjack/src/lib/formatting/lumberjack-format-log.spec.ts (hunk 3 above)
Comment (patelvimal) [reply libs/ngworker/lumberjack/src/lib/formatting/lumberjack-format-log.spec.ts:15]: This is showing as a warning not error. However there are many warning lint is showing should be fixed those as well?

File: libs/ngworker/lumberjack/console-driver/src/lib/configuration/lumberjack-console-driver-root.module.ts (hunk 1 above)
Comment (patelvimal) [reply libs/ngworker/lumberjack/console-driver/src/lib/configuration/lumberjack-console-driver-root.module.ts:42]: DId not understand this comment, can you ellaborate what needs to be done? Initally it was disabled that's why I disabled in eslint as well.

File: libs/ngworker/lumberjack/http-driver/src/lib/configuration/lumberjack-http-driver-root.module.ts (hunk 2 above)
Comment (patelvimal) [reply libs/ngworker/lumberjack/http-driver/src/lib/configuration/lumberjack-http-driver-root.module.ts:48]: ok, let me apply those rules as well.

**Comment by NachoVazquez** [comment]
Hey @patelvimal !

Thanks again for the help. I'm actually doing the same migration in another branch where I'm migrating to Angular 13

However, I prefer to have two separate PR and then I can rebase on yours.

Here is a list of the files that I think you are still missing to migrate:

### Sonar

- `tools/scripts/configure-report-paths.mjs`
- `sonar-project.properties`

### Lumberjack App

- `apps/lumberjack-app/src/app/app-logger.service.ts`
- `apps/lumberjack-app/src/app/app.components.ts`

### Internal

- `libs/internal/test-util/src/lib/error-throwing-driver/error-throwing-driver.module.ts`
- `libs/internal/test-util/src/lib/noop-driver/noop-driver.module.ts`
- `libs/internal/test-util/src/lib/object-driver/object-driver.module.ts`
- `libs/internal/test-util/src/lib/spy-driver/spy-driver.module.ts`
- `libs/internal/test-util/src/lib/spy-driver/spy.driver.ts`
- `libs/internal/test-util/src/lib/type-checking/is-class.ts`

Although, some of the files I marked are affected by rules that I didn't override. Just verify that lints are passing

**Review by NachoVazquez — CHANGES_REQUESTED** [review]

File: .eslintrc.json [hunk 4]
```diff
@@ -0,0 +1,25 @@
+{
+  "root": true,
```
[the remark this answers is not in the document]
Comment (patelvimal) [reply .eslintrc.json:2]: @NachoVazquez I have updated the eslint configuration based on this, please have a look into the PR.

**Comment by patelvimal** [comment]
> Hey @patelvimal !
> 
> Thanks again for the help. I'm actually doing the same migration in another branch where I'm migrating to Angular 13
> 
> However, I prefer to have two separate PR and then I can rebase on yours.
> 
> Here is a list of the files that I think you are still missing to migrate:
> 
> ### Sonar
> * `tools/scripts/configure-report-paths.mjs`
> * `sonar-project.properties`
> 
> ### Lumberjack App
> * `apps/lumberjack-app/src/app/app-logger.service.ts`
> * `apps/lumberjack-app/src/app/app.components.ts`
> 
> ### Internal
> * `libs/internal/test-util/src/lib/error-throwing-driver/error-throwing-driver.module.ts`
> * `libs/internal/test-util/src/lib/noop-driver/noop-driver.module.ts`
> * `libs/internal/test-util/src/lib/object-driver/object-driver.module.ts`
> * `libs/internal/test-util/src/lib/spy-driver/spy-driver.module.ts`
> * `libs/internal/test-util/src/lib/spy-driver/spy.driver.ts`
> * `libs/internal/test-util/src/lib/type-checking/is-class.ts`
> 
> Although, some of the files I marked are affected by rules that I didn't override. Just verify that lints are passing

Thanks @NachoVazquez ,  I have  checked again and all the above files are passing the ESLint rules.

**Comment by NachoVazquez** [comment]
Hi @patelvimal!

I think that the sonar files are still using tslint. If you look inside the files you will notice that there are references to using tslint as the linter.

Performing a search inside both of the files mentioned above will point you in the right direction.

Thanks again!

**Comment by patelvimal** [comment]
Hi @NachoVazquez 
Thanks for pointing out that, you were right. I have removed all those references and also some tslint packages as well.
Please review it and let me know if any changes required.

**Review by NachoVazquez — APPROVED** [review]

**Comment by patelvimal** [comment]
Thanks @NachoVazquez for review .

**Pull request merged**